src/countdown-timer/countdown-timer.tsx

```diff
@@ -0,0 +1,172 @@
+import debounce from "lodash/debounce";
+import { useEffect, useMemo, useRef, useState } from "react";
+import { useMediaQuery } from "react-responsive";
+import { useTimer } from "../util/use-timer";
+import { CountdownTimerProps } from "./types";
+import {
+    Countdown,
+    PositionStyle,
+    Time,
+    TimeLeft,
+    Wrapper,
+} from "./countdown-timer.style";
+import { TimeHelper } from "../util/time-helper";
+import { ClockIcon } from "@lifesg/react-icons";
+import { useIntersectionObserver } from "../util/use-intersection-observer";
+import { useEventListener } from "../util/use-event-listener";
+import { MediaWidths } from "../spec/media-spec";
+
+export const CountdownTimer = ({
+    className,
+    timer: _timer,
+    notifyTimer,
+    offset,
+    show,
+    "data-testid": testId,
+    onDuration,
+    onNotify,
+    onFinish,
+}: CountdownTimerProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const { minutes = 0, seconds = 0 } = _timer;
+    const notifyMinutes = notifyTimer?.minutes || 0;
+    const notifySeconds = notifyTimer?.seconds || 0;
+    const notifyTiming = notifyMinutes * 60 + notifySeconds;
+    const totalSeconds = minutes * 60 + seconds;
+
+    const wrapperRef = useRef<HTMLDivElement>();
+    const stickyRef = useRef<HTMLDivElement>();
+
+    const [scrolling, setScrolling] = useState<boolean>(false);
```

Comment:
pending confirmation with UX if this is needed

Reply:
Thanks spotted this issue. yeah, it appears shadow only in floating element 🤕. Thank you